content/blog/LFX-Mentorship-Blog.md

```diff
@@ -0,0 +1,425 @@
+
+**No wal**
+
+```trickle
+define script counter
+script
+  let count = match state==null of
+    case true =>
+      0
+    default =>
+      state.count + 1
+  end;
+
+  let state = {"count": count};
+  {"command": "send_message", "project":"<project-id>", "topic":"<topic-name>", "data": merge event of {"count": state.count} end, "ordering_key": "<ordering-key>"}
+end;
+
+create script counter;
+
+select event from in into counter;
+select event from counter into out;
+```
+
+**Transient-wal**
+
+```trickle
+define qos::wal operator in_memory_wal
+with
+  read_count = 20,
+  max_elements = 1000, # Capacity limit of 1000 stored events
+  max_bytes = 10485760 # Capacity limit of 1MB of events
+end;
+
+create operator in_memory_wal;
+
+define script counter
+script
+  let count = match state==null of
+    case true =>
+      0
+    default =>
+      state.count + 1
+  end;
+
+  let state = {"count": count};
+  {"command": "send_message", "project":"<project-id>", "topic":"<topic-name>", "data": merge event of {"count": state.count} end, "ordering_key": "<ordering-key>"}
+end;
+
+create script counter;
+
+select event from in into counter;
+select event from counter into in_memory_wal;
+select event from in_memory_wal into out;
+```
+
+**Persistent-wal**
+
+```trickle
+define qos::wal operator on_disk_wal
+with
+  read_count = 20,
+  max_elements = 1000, # Capacity limit of 1000 stored events
+  max_bytes = 10485760 # Capacity limit of 1MB of events
+end;
+create operator on_disk_wal;
+
+define script counter
+script
+  let count = match state==null of
+    case true =>
+      0
+    default =>
+      state.count + 1
+  end;
+
+  let state = {"count": count};
+  {"command": "send_message", "project":"<project-id>", "topic":"<topic-name>", "data": merge event of {"count": state.count} end, "ordering_key": "<ordering-key>"}
+end;
+
+create script counter;
+
+select event from in into counter;
+select event from counter into on_disk_wal;
+select event from on_disk_wal into out;
+```
+
+In the **inbound** trickle, we have the validation logic as follows:
+
+```trickle
+define script validate
+script
+  match state == null of
+    case true =>
+      let valid = match event.data.count == 0 of 
+        case true =>
+          true      
+        default =>
+          false
+      end,
+      let state = {"prev": event.data.count},
+      {"response": event, "valid": valid} 
+      
+    default =>
+      let valid = match state.prev + 1 == event.data.count of 
+        case true =>
+          true
+        default =>
+          false
+      end,
+      let state = {"prev": event.data.count},
+      
+      {"response":event, "valid": valid}
+  end;
+end;
+
+create script validate;
+
+select {"data": event, "meta": $} from in into validate;
+select event from validate into out;
+```
+
+For the validation, we run the source (onramp) and sink (offramp) separately.
+
```

Comment:
What about adding a final sentence here about the result of the validation testing?

Reply:
the sequence was preserved in all the cases. - I observed duplicates and messages being missed after restarting tremor in case of transient wal (when testing in poor network) but that was not the case with persistent wal - But I wasn't sure if it's nice to add this